### ISSUE TITLE
Cop idea: detect misspelling of constructor

### ISSUE THREAD

**Issue by viralpraxis** [rubocop/rubocop#13980]
## Is your feature request related to a problem? Please describe.

First of all, I'm not entirely sure if this fits well with RuboCop. However, I'd be happy to hear peoples opinions on this.

Sometimes (though rarely), I encounter bugs caused by misspelling the `initialize` method name. The last time I saw this issue, it was in a fairly popular repository, where it [went unnoticed](https://github.com/palkan/isolator/commit/cee5abb2c69e0c2d8b0405d1175052f7c0ae13f4#diff-947844fbcc8442ed7929279a8ce1dc9222832ef808043b7c23a07cb2958cecaeR23) for 5 years and gave me some headaches while debugging an application bug.

The word `initialize` is definitely tricky to spell (especially at fridays) correctly. A quick GitHub code search shows some occurrences of this exact typo: https://github.com/search?q=initilize+language%3Aruby&type=code

What if we could detect such typos? A proper implementation -- something like
`name.size == "initialize".size && levenshtein_distance(name, "initialize") == 1` would likely produce no false positives and be practically useful.

Again, I'm not sure if this fits within RuboCop's scope, or even which department it would belong to.

## Additional context

It usually happens when you have zero-arity constructor or a constructor where each argument has a default value, and the class instantination uses superclass constructor.

```ruby
class A
  def initalize(a = 1)
    @a  = a
  end
end

a = A.new
p ainstance_variable_get(:@a) # nil
```
